brainscore/benchmarks/lee2020.py

```diff
@@ -0,0 +1,208 @@
+        '''
+        candidate_model.start_recording(recording_target='IT', time_bins=[(70, 170)])
+        candidate_assembly = candidate_model.look_at(self._stimulus_set)
+        candidate_assembly = make_static(candidate_assembly)
+
+        candidate_statistic = []
+        for window in sample_array_locations(candidate_assembly.neuroid, self.num_sample_arrs, self._array_size_mm):
+            array_statistic = sample_response_corr_vs_dist(candidate_assembly[window],
+                                                           int(self.bootstrap_samples / self.num_sample_arrs))
+            candidate_statistic.append(array_statistic)
+
+        self._target_statistic = self._target_statistic_fn()
+        self._ceiling_func = lambda: InterIndividualStatisticsCeiling(self._score)(
+            self._target_statistic)  # TODO integrate ceiling
+
+        score = self._score(np.hstack(self._target_statistic), np.hstack(candidate_statistic))
+        score.attrs['target_statistic'] = self._target_statistic
+        score.attrs['candidate_statistic'] = candidate_statistic
+
+        return score
+
+    def create_tissue_response_correlation_target(self):
+        '''
+        :return: list of [pairwise distances, pairwise response correlations], each item is another animal/array
+            combination from the MajajHong Assembly
+        '''
+        target_statistic = []
+        for animal in sorted(list(set(self._target_assembly.neuroid.animal.data))):
+            for arr in sorted(list(set(self._target_assembly.neuroid.arr.data))):
+                sub_assembly = self._target_assembly.sel(animal=animal, arr=arr)
+                bootstrap_samples_sub_assembly = int(self.bootstrap_samples * (sub_assembly.neuroid.size /
+                                                                               self._target_assembly.neuroid.size))
+                sub_assembly_statistic = sample_response_corr_vs_dist(sub_assembly, bootstrap_samples_sub_assembly,
+                                                                      self._neuroid_reliability)
+                target_statistic.append(sub_assembly_statistic)
+
+        return target_statistic
+
+
+def sample_response_corr_vs_dist(assembly, num_samples, neuroid_reliability=None):
+    '''
+    1. Samples random pairs from the assembly
+    2. Computes distances for all pairs
+    3. Computes the response correlation between items of each pair
+    (4. Ceils the response correlaions)
+    :param assembly: NeuroidAssembly without stimulus repetitions
+    :param num_samples: how many random pair you want to be sampled out of the data
+    :param neuroid_reliability: if not None: expecting Score object containing reliability estimates of all neuroids
+    :return: [distance, pairwise_correlation_of_neuroids], pairwise correlations can be ceiled
+    '''
+    rng = np.random.default_rng()

```

Comment:
I would use a random state here (and whenever you use randomness) to ensure perfect reproducibility:
`rng = RandomState(0)`. Need to `from numpy.random import RandomState`, and 0 can be any number to seed the generator.

Reply:
if I am [understanding correctly for numpy > 1.17](https://numpy.org/doc/stable/reference/random/generator.html#numpy.random.Generator) the preferred way is to use `np.random.default_rng()` to which you can pass a seed making everything reproducible. This should result in the desired behavior, right?

Question for this: In case you like that solution, do you think it makes sense to have the random generator being part of each benchmark instance? Would be more convenient but also if people play around and call things in different orders, random numbers are not going to be  the same.

Reply:
oh I didn't know about the new Generators, neat. 
I think where you have the `rng` now is fine, all that is relevant is that if you call the benchmark with the same parameters (aka model), it should yield the exact identical result.